src/main/java/edu/ie3/datamodel/io/source/sql/SqlLoadProfileSource.java

```diff
@@ -0,0 +1,205 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.measure.quantity.Energy;
+import javax.measure.quantity.Power;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import tech.units.indriya.ComparableQuantity;
+
+/**
+ * Sql source for {@link LoadProfileTimeSeries}.
+ *
+ * @param <P> type of load profile
+ * @param <V> type of load values
+ */
+public class SqlLoadProfileSource<P extends LoadProfile, V extends LoadValues>
+    extends LoadProfileSource<P, V> {
+  protected static final Logger log = LoggerFactory.getLogger(SqlTimeSeriesSource.class);
+  private final SqlDataSource dataSource;
+  private final String tableName;
+
+  private final LoadProfileMetaInformation metaInformation;
+  private final P loadProfile;
+
+  // General fields
+  private static final String WHERE = " WHERE ";
+  private static final String LOAD_PROFILE = "load_profile";
+
+  /**
+   * Queries that are available within this source. Motivation to have them as field value is to
+   * avoid creating a new string each time, bc they're always the same.
+   */
+  private final String queryFull;
+
+  private final String queryTime;
+
+  public SqlLoadProfileSource(
+      SqlDataSource dataSource,
+      LoadProfileMetaInformation metaInformation,
+      Class<V> entryClass,
+      LoadProfileFactory<P, V> entryFactory) {
+    super(entryClass, entryFactory);
+    this.dataSource = dataSource;
+
+    this.tableName = "load_profiles";
+    this.metaInformation = metaInformation;
+    this.loadProfile = entryFactory.parseProfile(metaInformation.getProfile());
+
+    String dbTimeColumnName =
+        dataSource.getDbColumnName(entryFactory.getTimeFieldString(), tableName);
+
+    this.queryFull = createQueryFull(dataSource.schemaName, tableName);
+    this.queryTime = createQueryForTime(dataSource.schemaName, tableName, dbTimeColumnName);
+  }
+
+  public SqlLoadProfileSource(
+      SqlConnector connector,
+      String schemaName,
+      DatabaseNamingStrategy namingStrategy,
+      LoadProfileMetaInformation metaInformation,
+      Class<V> entryClass,
+      LoadProfileFactory<P, V> entryFactory) {
+    this(
+        new SqlDataSource(connector, schemaName, namingStrategy),
+        metaInformation,
+        entryClass,
+        entryFactory);
+  }
+
+  @Override
+  public void validate() throws ValidationException {
+    validate(entryClass, () -> dataSource.getSourceFields(tableName), entryFactory);
+  }
+
+  // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+
+  @Override
+  public LoadProfileTimeSeries<V> getTimeSeries() {
+    Set<LoadProfileEntry<V>> entries = getEntries(queryFull, ps -> {});
+    return entryFactory.build(metaInformation, entries);
+  }
+
+  @Override
+  public List<ZonedDateTime> getTimeKeysAfter(ZonedDateTime time) {
+    return List.of(time.plusMinutes(15));
+  }
```

Comment:
This would only work correct, when the time series are always in quarter hours steps, right? Should this get adapted to (optional) provide the step size and only take 15 minutes if none is provided?